capstone/fitness/src/main/java/com/google/sps/servlets/CalendarServlet.java

```diff
@@ -29,71 +33,89 @@
+        String nextDayStartTime = "T11:00:00+00:00";
+        String nextDayEndTime = "T23:00:00+00:00";
+        String minRCF3339 = nextDayOfMonth + nextDayStartTime;
+        String maxRCF3399 = nextDayOfMonth + nextDayEndTime;
+        DateTime minSpan = new DateTime(minRCF3339);
+        DateTime maxSpan = new DateTime(maxRCF3399);
 
-    // Use the scheduler to schedule one exercise per day. 
-    int y = 0;
+        // Use the scheduler to schedule one exercise per day. 
+        int y = 0;
+    
+        for (int x = 0; x < daysAvailable; x += timesPerWeek){
+          if (y >= exercises.size()){
+            break;
+            }
+          else{
+            List<Event> currentlyScheduledEvents = this.getEventsInTimespan(minSpan, maxSpan);
+            // User scheduler to get free time and return corresponding event. 
+            Event exerciseEvent = scheduler.getFreeTime(minSpan, maxSpan, currentlyScheduledEvents);
+            // Event name and description match the user's workout name and type of workout.
+            exerciseEvent.setSummary(APPLICATION_NAME + ": " + exercises.get(y));
+            exerciseEvent.setDescription(type);
+            // Set event color depending on the type of workot. 
+            if (type.equals("lifting")){
+              exerciseEvent.setColorId(liftingColorId);
+            }
+            else{
+              exerciseEvent.setColorId(runningColorId);
+            }
+            this.insertEvent(exerciseEvent);
+            
+            // Store each event's eventID in datastore for display later.
+            String eventDescription = type + " at " + exerciseEvent.getStart().getDateTime();
+            DataHandler.addEventID(user,eventDescription);
 
-    for (int x = 0; x < daysAvailable; x += timesPerWeek){
-      if (y >= exercises.size()){
-        break;
-        }
-      else{
-        List<Event> currentlyScheduledEvents = this.getEventsInTimespan(minSpan, maxSpan);
-        Event exerciseEvent = scheduler.getFreeTime(minSpan, maxSpan, currentlyScheduledEvents);
-        exerciseEvent.setSummary(APPLICATION_NAME + ": " + exercises.get(y));
-        exerciseEvent.setColorId(colorId);
-        // TODO (@piercedw) : Store each event's eventID in datastore for display later.
-        this.insertEvent(exerciseEvent);
-        // Increment minSpan and maxSpan by one day. 
-        minSpan = new DateTime(minSpan.getValue() + (Time.millisecondsPerDay * timesPerWeek));
-        maxSpan = new DateTime(maxSpan.getValue() + (Time.millisecondsPerDay * timesPerWeek));
-        y++;
-    }
-    }
-    String id = this.getCalendarId();
-    response.sendRedirect("/calendar.html?calendarId=" + id); 
-  }
+            // Increment minSpan and maxSpan by one day. 
+            minSpan = this.incrementDay(minSpan);
+            maxSpan = this.incrementDay(maxSpan);
+            y++;}}
+        scheduledLength++; }
+      // Store calendar ID. 
+      String id = this.getCalendarId();
```

Comment:
Maybe just directly getCalenderId when setting DataHandler since you only use this value once?